commit e16563238a07592944fd94cd42cae07e767d8b78
Author: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>
Date:   Tue Nov 27 16:59:10 2018 +0100

    BCM2710_DT: fix gpio expander bindings
    
    The upstreamed driver for the GPIO expander expects to be a children of
    the "firmware" node.
    
    The patch also removes the "firmware" phandle as it's useless.
    
    Signed-off-by: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>

diff --git a/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts b/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
index fe1e2d2daf62..898ab48479e3 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
@@ -83,12 +83,11 @@
 	brcm,overclock-50 = <0>;
 };
 
-&soc {
+&firmware {
 	expgpio: expgpio {
 		compatible = "raspberrypi,firmware-gpio";
 		gpio-controller;
 		#gpio-cells = <2>;
-		firmware = <&firmware>;
 		status = "okay";
 	};
 };
diff --git a/arch/arm/boot/dts/bcm2710-rpi-3-b.dts b/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
index 36c567517186..61ecb46b9c06 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
@@ -92,11 +92,13 @@
 		status = "okay";
 	};
 
+};
+
+&firmware {
 	expgpio: expgpio {
 		compatible = "raspberrypi,firmware-gpio";
 		gpio-controller;
 		#gpio-cells = <2>;
-		firmware = <&firmware>;
 		status = "okay";
 	};
 };
diff --git a/arch/arm/boot/dts/bcm2710-rpi-cm3.dts b/arch/arm/boot/dts/bcm2710-rpi-cm3.dts
index 2ed05b73e824..9490177443a5 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-cm3.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-cm3.dts
@@ -53,11 +53,13 @@
 		status = "okay";
 	};
 
+};
+
+&firmware {
 	expgpio: expgpio {
 		compatible = "raspberrypi,firmware-gpio";
 		gpio-controller;
 		#gpio-cells = <2>;
-		firmware = <&firmware>;
 		status = "okay";
 	};
 };
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1319_e16563238a07592944fd94cd42cae07e767d8b78.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1319_e16563238a07592944fd94cd42cae07e767d8b78.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
